### DISCUSSION TITLE
Add Azd support for the simple environment

### DISCUSSION THREAD

**Discussion by sonwan2020** [Azure-Samples/java-microservices-aca-lab#16] [General]
Target:
Single `azd up` command to deploy the petclinic solution to Azure Container Apps environment

This task will auto finish the steps of lab 1/2/3/4

Some flags to control:
* external network or internal network

With external network, we finish the steps of 1/2/3/4 in one command
With internal support, we can quickly jump to the more secure solution starting from `'Lab 5: Protect endpoints using Web Application Firewalls`

Note:
* The steps of other labs will discussed later

**Comment by vermegi** [comment]
I really like this idea. The ask has been there for a while to be able to quickly set up the 'end state'. A couple of thoughts: 

- what about the config repo? A lot of the setup working properly is dependent on what is present in your config repo. How would we solve for this? 
- another thought/idea that popped up in the past was adding a lab on 'automation'. At the time we were thinking GH actions in combo with Bicep. It would then rather be explaining how they would get at the this rather then providing a 1 button/statement deploy that does 'magic'. Curious to hear your thoughts on this. 
- Do you have any idea of the effort for azd support?

**Comment by sonwan2020** [comment]
Thanks for supporting this idea, actually now I am working on this task.

* we give a AZD solution for this sample, we need some more technical sections to explain what happens to different config repo.
* We may use another thread to discuss the GH https://github.com/Azure-Samples/java-microservices-aca-lab/discussions/19
* I will do this step by step

Currently I am not an expert on azd & bicep area, I try to get some time line later some time.

* minimal version:
rebuild container environment & apps, replace the lab 5 section 4 operations to destroy & recreate steps.

* the core version
build the entire lab steps 1/2/3/4 with internal network
for those want to dive into the security parts, we offer a core environment in a hour, users can skip the core labs parts

* full secure version
  we can deploy the solution with secure components.
  Some options to control the parts

**Comment by vermegi** [comment]
I should have time later this week and hopefully next week to add to the azd lab. It would be nice if we can put additional explanation steps in it for the learner.

**Comment by vermegi** [comment]
@sonwan2020 could we also add the creation of the ACR repository to the bicep templates? In combination with the import statement, we could then have everything in 1 go without needing any extra steps. 

Import in Bicep: https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep

**Comment by sonwan2020** [comment]
Yes, I will create another discussion about this.


Regards,

Songbo



________________________________

**Comment by vermegi** [comment]
Also, it would be nice if we could 'pull apart' this automation lab in the following (with more explanation as well):

- automate environment creation with bicep templates 
  - add explanation on how the bicep files are build up 
  - add steps for az deployment group create
-  automate environment creation with azd
   - add explanation for azure.yml file
- automatically deploy with GH actions

**Comment by sonwan2020** [comment]
Agree.
Let's discuss the GH actions in a new thread


Regards,

Songbo



________________________________
